lib/internal/fs/streams.js

```diff
@@ -100,98 +67,156 @@ const FileHandleOperations = (handle) => {
   };
 };
 
-function close(stream, err, cb) {
-  if (!stream.fd) {
-    // TODO(ronag)
-    // stream.closed = true;
-    cb(err);
-  } else {
-    stream[kFs].close(stream.fd, (er) => {
-      stream.closed = true;
-      cb(er || err);
-    });
-    stream.fd = null;
+function FileSystemStream(path, options) {
+
+  this[kFs] = options.fs || fs;
+
+  if (typeof this[kFs].open !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('options.fs.open', 'function',
+                                   this[kFs].open);
+  }
+
+  if (typeof this[kFs].close !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('options.fs.close', 'function',
+                                   this[kFs].close);
   }
-}
 
-function importFd(stream, options) {
-  stream.fd = null;
   if (options.fd) {
     if (typeof options.fd === 'number') {
       // When fd is a raw descriptor, we must keep our fingers crossed
       // that the descriptor won't get closed, or worse, replaced with
       // another one
       // https://github.com/nodejs/node/issues/35862
-      stream.fd = options.fd;
+      this.fd = options.fd;
     } else if (typeof options.fd === 'object' &&
       options.fd instanceof FileHandle) {
       // When fd is a FileHandle we can listen for 'close' events
       if (options.fs)
         // FileHandle is not supported with custom fs operations
         throw new ERR_METHOD_NOT_IMPLEMENTED('FileHandle with fs');
-      stream[kHandle] = options.fd;
-      stream.fd = options.fd.fd;
-      stream[kFs] = FileHandleOperations(stream[kHandle]);
-      stream[kHandle][kRef]();
-      options.fd.on('close', FunctionPrototypeBind(stream.close, stream));
+      this[kHandle] = options.fd;
+      this.fd = options.fd.fd;
+      this[kFs] = FileHandleOperations(this[kHandle]);
+      this[kHandle][kRef]();
+      options.fd.on('close', FunctionPrototypeBind(this.close, this));
     } else
       throw ERR_INVALID_ARG_TYPE('options.fd',
                                  ['number', 'FileHandle'], options.fd);
+  } else {
+    this.fd = null;
+    this.path = toPathIfFileURL(path);
+  }
+
+  options.autoDestroy =
+    options.autoClose === undefined ? true : options.autoClose;
+
+  this.mode = options.mode === undefined ? 0o666 : options.mode;
+
+  this.start = options.start;
+  this.pos = options.start;
+  this.closed = false;
+  this[kIsPerformingIO] = false;
+
+  if (this.start !== undefined) {
+    validateInteger(this.start, 'start', 0);
   }
 }
 
+FileSystemStream.prototype[kClose] = function close(err, cb) {
+  if (this.fd) {
+    this[kFs].close(this.fd, (er) => {
+      this.closed = true;
+      cb(er || err);
+    });
+    this.fd = null;
+  } else {
+    // TODO(ronag)
+    // stream.closed = true;
+    cb(err);
+  }
+};
```

Comment:
This doesn't need a symbol. Just make it a function taking stream as first param.